python/src/equistore/operations/reduce.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+
+
+from equistore import Labels, TensorBlock, TensorMap
+
+from . import _dispatch
+
+# TODO maybe add a way to reduce only the rows
+#  with sample[sample_name] = idx
+# for example not summing over all the sample
+# that have the same 'structure'
+# but only those with 'structure' = 0
+
+
+def _reduction_sample_block(block: TensorBlock, sample_name: list) -> TensorBlock:
```

Comment:
```suggestion
def _reduction_sample_block(block: TensorBlock, sample_names: list) -> TensorBlock:
```

since there are multiple names

Reply:
also, the right type should be

```suggestion
def _reduction_sample_block(block: TensorBlock, sample_name: List[str]) -> TensorBlock:
```